Server/mods/deathmatch/logic/CMapManager.cpp

```diff
@@ -592,6 +592,7 @@ void CMapManager::SpawnPlayer(CPlayer& Player, const CVector& vecPosition, float
     }
 
     // Update the player data
+    Player.SetPlayerSpawned(true);
```

Comment:
I see that SetPlayerSpawned is called just ones. Should't there be a place where you call `SetPlayerSpawned(false)`?

Reply:
> I see that SetPlayerSpawned is called only ones. Should't there be a place where you call `SetPlayerSpawned(false)`?

You can't despawn a player

Reply:
> > I see that SetPlayerSpawned is called only ones. Should't there be a place where you call `SetPlayerSpawned(false)`?
> 
> You can't despawn a player

What will happened when player is died then?

Reply:
Actually, maybe I should just update `SetSpawned` to only work for spawning player and retain `SetDead` for killing/reviving players/peds

Reply:
> > What will happened when player is died then?
> 
> When player dies, `m_bHasPlayerSpawned` will change to `false` (indicating that the player has been despawned) but the player itself wont despawn, he will retain every property of a spawned player. Its kinda confusing tbf. Dead players should be dead, not despawned.

But how exactly the value of m_bHasPlayerSpawned can become false? There's no place where it's set to false in your code.

Reply:
> > > What will happened when player is died then?
> > 
> > 
> > When player dies, `m_bHasPlayerSpawned` will change to `false` (indicating that the player has been despawned) but the player itself wont despawn, he will retain every property of a spawned player. Its kinda confusing tbf. Dead players should be dead, not despawned.
> 
> But how exactly the value of m_bHasPlayerSpawned can become false? There's no place where it's set to false in your code.

Oops, sorry. I meant `m_bSpawned` (the original variable that changes when player dies)

`m_bHasPlayerSpawned` should only be set to false when player joins and leaves.
Since there is no direct way to despawn a player, it will never be set to false once a player has been spawned.

Reply:
> Actually, maybe I should just update `SetSpawned` to only work for spawning player and retain `SetDead` for killing/reviving players/peds? That would make more sense, no?

Yeah. I think `m_bSpawned` could be renamed to m_bDead (with the corresponding refactor, of course). When player connects m_bDead is false, when he's dead m_bDead is true (obviously), and when he's spawned after it m_bDead is false again. In this case we can avoid the collision and there will no any confusion.